flytekit/core/type_engine.py

```diff
@@ -55,6 +55,30 @@
 DEFINITIONS = "definitions"
 
 
+class BatchSize:
+    """
+    This is used to annotate a FlyteDirectory when we want to download/upload the contents of the directory in batches. For example,
+
+    def t1(Annotated[FlyteDirectory, BatchSize(10)]) -> Annotated[FlyteDirectory, BatchSize(100)]:
+        ...
+    """
+
+    def __init__(self, val: int):
+        self._val = val
+
+    @property
+    def val(self) -> int:
+        return self._val
+
+
+def get_batch_size(t: Type) -> Optional[int]:
+    if is_annotated(t):
+        for annotation in get_args(t)[1:]:
+            if isinstance(annotation, BatchSize):
+                return annotation.val
+    return None
```

Comment:
should we set the default batch size to 100 or something else? default behavior is to read all the files in the directory.

Reply:
cc @wild-endeavor

Reply:
according to the [docs](https://github.com/fsspec/filesystem_spec/blob/master/fsspec/asyn.py#L218-L221) `fsspec` already tries to find a reasonable batch size if we pass `batch_size=None`.

Reply:
nice, even better. didn't see it, thanks!